test/models/config_test.rb

```diff
@@ -3,6 +3,31 @@
 class ConfigTest < ActiveSupport::TestCase
   before { @config = create :config }
 
+  describe 'callbacks' do
+    it 'should clean URLs before save' do
+      # doesn't touch https
+      c = Config.find_or_create_by(config_key: 'resources_url')
```

Comment:
I have a suggestion for how to simplify this test a little bit - why don't we:

* move the meat of the cleaning into `UriService#clean_uri`
* do the heavy duty testing of the cleaning logic there (L20 on down, but just testing various strings)
* change this test to something that does a spot check to make sure that urls are getting cleaned when necessary (maybe the `http` -> `https`), and things are getting skipped appropriately like on L12 and L17)
* loop through each of the url options to repeat the test, to make sure they're all covered

what do you think?